fix(lint): Correct the cidrs for linux routing in cni

sayboras opened the pull request:
Related to
https://github.com/cilium/cilium/pull/11528#discussion_r425315197

Closes #11562

Signed-off-by: Tam Mach <sayboras@yahoo.com>

```release-note
Correct cidr input in linuxRouting.NewRoutingInfo
```

tklauser reviewed (changes requested):
It would be great if you could explain the reason behind this change (i.e. why it is safe to convert `cidrs` to a string slice and use it instead of `ipam.Cidrs` and why no CIDRs are lost in the process) in the commit message. Otherwise it will be hard to understand from the git log alone (without referring to the linked PR comments) why this change was made.

christarazi reviewed (changes requested):
~~The changes look good. Thanks for following up on this! One item to address for me.~~

christarazi commented on pkg/datapath/linux/routing/info.go:
```diff
@@ -56,7 +56,7 @@ func parse(gateway string, cidrs []string, macAddr string) (*RoutingInfo, error)
 		return nil, fmt.Errorf("invalid ip: %s", gateway)
 	}
 
-	if cidrs == nil || len(cidrs) == 0 {
```
Why was this nil check removed? It'll make this test fail: https://github.com/cilium/cilium/blob/84db4f63a7c99c472823b84c202196897272fb4a/pkg/datapath/linux/routing/info_test.go#L79

sayboras replied:
@christarazi thanks for your input. I enabled for _gosimple_ in my IDE by default, and basically len() for nil slices is defined as zero in go, so I don't think the above test is failing. I just verify by running the test locally as well.

Feel free to let me know otherwise.

christarazi replied:
Ah, I didn't know that!  :)

christarazi reviewed (changes requested):
The changes look good. Thanks for following up on this! One item to address for me.

sayboras committed:
fix(lint): Use coalesce CIDRs  for linux routing in cni

Currently, the coalesce cidrs was derived from ipam.CIDRs with the intention
of minimizing routing rules, however original full ipam.cidrs are passed
to routing rules instead.

This PR is to correct such discrepancy

Closes #11562

Fixes: b58dd147ed44 ("Refactor enirouting pkg to use interfaces")

Signed-off-by: Tam Mach <sayboras@yahoo.com>

aanm reviewed (changes requested):
Nice finding. Small nit can you add a 

```
Fixes: b58dd147ed44 ("Refactor enirouting pkg to use interfaces")
```
in the commit message?
